arm64: dts: rk3399: improve rock pi 4 emmc config

LZhaoM opened the pull request:
arm64: dts: rk3399: improve rock pi 4 emmc config
    
    This change is based on this commit[0].
    
    However, the property 'enable-strobe-pulldown' has already been set[1].
    We just need to set 'mmc-hs400-1_8v' and 'mmc-hs400-enhanced-strobe' in
    this change.
    
    For testing compactability, the following eMMCs were tested on Rock
    4B:
    
    Foresee 16G, 32G, 64G
    Samsung 16G, 32G, 128G
    Sandisk 8G, 16G
    
    [0]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=f720dd9b8b6d8b2160beda789429d5489ce8a099
    [1]: https://github.com/radxa/kernel/commit/b045498f5c40faf27485dcfa9429b8e73f9aa9ce
    
    Signed-off-by: Zhaoming Luo <luozhaoming@radxa.com>

RadxaYuntian commented on arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi:
```diff
@@ -711,9 +711,10 @@
 
 &sdhci {
-	max-frequency = <150000000>;
+	max-frequency = <384000000>;
```
https://www.rockchip.fr/Rockchip%20RK3399%20TRM%20V1.3%20Part2.pdf p605说了最大速率200M

RadxaYuntian reviewed (commented):
你有了解过这么改的原因么？
https://github.com/radxa/kernel/commit/a25a68af4e53134f120cfea9d4b5c9f71da275dc
https://github.com/radxa/kernel/commit/d71e69b9bc169244d8f16e788ce6950b0cd3337a
https://github.com/radxa/kernel/commit/52d3607db0de375aca8d48d3445f9120f59814db

LZhaoM commented:
> 你有了解过这么改的原因么？ [a25a68a](https://github.com/radxa/kernel/commit/a25a68af4e53134f120cfea9d4b5c9f71da275dc) [d71e69b](https://github.com/radxa/kernel/commit/d71e69b9bc169244d8f16e788ce6950b0cd3337a) [52d3607](https://github.com/radxa/kernel/commit/52d3607db0de375aca8d48d3445f9120f59814db)

@RadxaYuntian 
这些 @RadxaStephen 昨天有跟我说，然后让我基于目前的修改提交一个eMMC兼容性测试，提测我已经提交了。接下来怎么处理好些？

RadxaYuntian commented:
至少速率要修复成正确的值，然后提测。如果没问题再合并

LZhaoM committed:
arm64: dts: rk3399: improve rock pi 4 emmc config

This change is based on this commit[0].

However, the property 'enable-strobe-pulldown' has already been set[1].
We just need to set 'mmc-hs400-1_8v' and 'mmc-hs400-enhanced-strobe' in
this change.

For testing compactability, the following eMMCs were tested on Rock
4B:

Foresee 16G, 32G, 64G
Samsung 16G, 32G, 128G
Sandisk 8G, 16G

[0]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=f720dd9b8b6d8b2160beda789429d5489ce8a099
[1]: https://github.com/radxa/kernel/commit/b045498f5c40faf27485dcfa9429b8e73f9aa9ce

Signed-off-by: Zhaoming Luo <luozhaoming@radxa.com>

LZhaoM commented:
兼容性测试结果[0]：
|EMMC品牌及容量	|SD卡启动读写速度|作为系统盘读写速度|
|---|---|---|
|江波龙 16G|154 MB/s\|84.4 MB/s|150MB/s\|81.0 MB/s|
|江波龙 32G|150 MB/s\|123 MB/s|149 MB/s\|124 MB/s|
|江波龙 64G|154 MB/s\|105 MB/s|158 MB/s\|103 MB/s|
|三星 16G|153 MB/s\|55.5 MB/s|150 MB/s\|54.2 MB/s|
|三星 32G|163 MB/s\|105 MB/s|149 MB/s\|98.7 MB/s|
|三星 128G|164 MB/s\|132 MB/s|148 MB/s\|125 MB/s|
|闪迪 8G|166 MB/s\|42.6 MB/s|容量小 无法刷入镜像|
|闪迪 16G|165 MB/s\|29.0 MB/s|161 MB/s\|27.6 MB/s|
|闪迪 32G|NA|NA|

[0]: https://vamrs.feishu.cn/sheets/TywcsJoABhrk0ZtMoHacXxrInOd

RadxaYuntian commented on arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi:
```diff
@@ -713,7 +713,8 @@
 &sdhci {
 	max-frequency = <150000000>;
 	bus-width = <8>;
-	mmc-hs200-1_8v;
```
所以hs200是有问题的么？

LZhaoM replied:
https://lore.kernel.org/all/CAGb2v65M5Zv73UKwEtmK=-MgDuubjKFA=R-bxceh+uL3iPuG6Q@mail.gmail.com/
这里有提到`mmc-hs400-1_8v`implies`mmc-hs200-1_8v`。我们的这边的代码也是这么个意思。
https://github.com/radxa/kernel/blob/3f671654064b5efa8538748356d2c9bd3b754462/drivers/mmc/core/host.c#L393-L394
https://github.com/radxa/kernel/blob/3f671654064b5efa8538748356d2c9bd3b754462/drivers/mmc/core/host.c#L397-L398

RadxaYuntian reviewed (commented):
另外8g emmc可以用cli镜像吧，不过应该不需要重新测试恶劣

RadxaStephen commented:
Merged.
